rust/src/dcerpc/dcerpc.rs

```diff
@@ -364,6 +365,9 @@ impl DCERPCState {
             }
             self.tx_index_completed = index;
         }
+        if let Some(hdr) = &self.header {
+            tx.version = hdr.rpc_vers * 10 + hdr.rpc_vers_minor;
```

Comment:
Could this uint arithmetic overflow ? which would lead to debug assertion in rust

Reply:
We verify before we reach here that `hdr.rpc_vers  == 5 && hdr.rpc_vers_minor == 0 or 1` so I believe this should always stay in the limits. See: https://github.com/OISF/suricata/blob/master/rust/src/dcerpc/dcerpc.rs#L616

Do you see another path where this would overflow?

Reply:
Ok,

Why do we need this ` tx.version` then ? Can we not store just hdr.rpc_vers_minor if it is the only one changing ? Do we not store it already ?

Reply:
We do store it but in the header. Because of a design problem, header is not a part of the tx but the state so the quick fix was to expose whatever is needed from the header to the tx.. which happened to be just this one field..

Reply:
> Can we not store just hdr.rpc_vers_minor if it is the only one changing ? 

Great idea